netbox/core/models/jobs.py

```diff
@@ -208,11 +213,21 @@ def enqueue(cls, func, instance, name='', user=None, schedule_at=None, interval=
             name: Name for the job (optional)
             user: The user responsible for running the job
             schedule_at: Schedule the job to be executed at the passed date and time
+            rq_queue_name: Queue name to route the job to for processing
             interval: Recurrence interval (in minutes)
         """
         object_type = ObjectType.objects.get_for_model(instance, for_concrete_model=False)
-        rq_queue_name = get_queue_for_model(object_type.model)
-        queue = django_rq.get_queue(rq_queue_name)
+        if rq_queue_name is None:
```

Comment:
You can refactor this logic to avoid the duplicate code, something like:

```
queue = None
if rq_queue_name:
    queue = django_rq.get_queue(...
if not queue:
    rq_queue_name = get_queue_for_model(...
    queue = django_rq.get_queue(...
```

Reply:
Updated the code: https://github.com/netbox-community/netbox/commit/5676c4abadc928ba00943dcd99329173c67c01ee